### ISSUE TITLE
need of an integration with the ores of create

### ISSUE THREAD

**Issue by Limoncafeinomano** [Builderb0y/BigGlobe#37]
As a matter of the generation form that this mod has, and the form of generation of ores of the "create" mod in addition to its patches of other decorative blocks, of the "stone" type, neither of these two are generated, I have been reading and quite a few people who like this mod think it is a big inconvenience that the generation of ores from other mods is incompatible in "Big Globe", which is why I open this issue to request an improved integration, or a section in the "settings" documents. ", to make it easier to add new flower and structure generation mods, dependent on world generation.

thanks for read. 
had a good day.

**Comment by Builderb0y** [comment]
I'm already working on porting a bunch of my old compatibility data packs, including create. They will be released when I re-add the invite for my discord server to my modrinth page. Current plan is to do this on the 26'th, but I might do it sooner if my page views die down faster than expected.

**Comment by Endri1987-1234567890** [comment]
what about mekanism?

**Comment by Builderb0y** [comment]
I haven't done that one yet, but it doesn't sound too difficult. It just has osmium ore, right?

**Comment by Endri1987-1234567890** [comment]
also tin, urranium, flourite, lead

**Comment by Builderb0y** [comment]
Alright. I guess the mod added some stuff since the last time I used it back in 1.7.10. Still, as long as it's just ores, it should be fairly easy to make a compatibility pack for.

**Comment by DeathByDaylight** [comment]
Is there some documentation or tutorial around how to write one of these compatibility data packs? My knowledge of this is too limited to freestyle it and I have had no success, but with a little bit of a template I don't imagine it would be too difficult to write one. Waiting till the 26th is fine if its too much of a hassle, but I was thinking it might let others try and get support working for more niche mods if there was a guide on how to get these data packs working.

**Comment by Builderb0y** [comment]
All of the documentation I have written so far can be found [here](https://github.com/Builderb0y/BigGlobe/tree/scriptable-generators/scripting%20docs). Ores in particular have a page [here](https://github.com/Builderb0y/BigGlobe/blob/scriptable-generators/scripting%20docs/files/data/modid/worldgen/configured_feature/Ores.md).

**Comment by DeathByDaylight** [comment]
Thanks, that's useful. The data pack I wrote isn't throwing any errors, but its not having any effect on terrain generation. In the pack I have effectively copied over the configured feature and tags folders, and added zinc in to both (The configured feature for zinc is a copy of iron), but the feature dispatcher doesn't seem to use these edited files. Are there any example files you have lying around? My attempt at the pack is attached.
[DatapackForBigGlobe.zip](https://github.com/user-attachments/files/17402606/DatapackForBigGlobe.zip)

**Comment by Builderb0y** [comment]
Odd. That pack looks "correct" to me, ignoring how many unnecessary files it has in it. Are you sure you installed the data pack in your world correctly?

**Comment by DeathByDaylight** [comment]
When creating the world, I open the data packs folder and copy that file over, the only difference being that its not zipped, and when I enable the pack it reads "validating data pack", so i assume its being installed correctly. My testing has just been flying around the caves in spectator, but I'm seeing a lot of iron and no zinc.

**Comment by Builderb0y** [comment]
Very strange. Not sure what the issue would be then.

**Comment by DeathByDaylight** [comment]
Ok, I'll have to do some more fiddling. For readabilities sake, what files can i remove? Also, could you attach one of the out of date compatibility packs to the thread so I can try and spot any differences that might lead to mine not working?

**Comment by Builderb0y** [comment]
You know what? I'll just post the updated create pack [here](https://github.com/user-attachments/files/17446664/Big.Globe.V4.5.0.Create.0.5.1-f-build.1417.Compatibility.zip) ahead of schedule. It adds zinc ore, and create's custom stone types as rock layers. Meanwhile, I've also decided to postpone the re-opening of my discord server due to 1.21.2 releasing soon, mostly because I don't want people joining to ask for ETA's. The new plan is to start porting 1 week after 1.21.2 is released (just to give mojang time to release hotfixes if necessary), and then re-open my discord server once porting is done.

Also to answer your question, you can remove any files that you didn't edit.

**Comment by DurekuRyuu** [comment]
> All of the documentation I have written so far can be found [here](https://github.com/Builderb0y/BigGlobe/tree/scriptable-generators/scripting%20docs). Ores in particular have a page [here](https://github.com/Builderb0y/BigGlobe/blob/scriptable-generators/scripting%20docs/files/data/modid/worldgen/configured_feature/Ores.md).

These first file is no longer there, and I only found the ores file [Here](https://github.com/Builderb0y/BigGlobe/blob/3cde1967f65bd186841ea8f0c32b0a3b833fa9af/docs/files/data/modid/worldgen/configured_feature/Ores.md?plain=1). Updated last week so i'm guessing it's up to date, I tried using the 4.5.0 Create compatibility datapack with 3.12.0 on 1.19.2 fabric with no success. I'm guessing this is due to a change in datapack formatting on mojang's end between versions? Nothing specifies a specific mod version aside from the filename so I don't know if there are any bandaid fixes I could apply to use this in my 1.19.2 modpack.

Ideally I'd also like to create compatibility datapacks for a number of things:

- Botania
- Create
- Create New Age
- Clockwork
- Valkyrien Skies
- Yung's mods for fabric on 1.19.2 (maybe)
- Dimensional Doors
- Terralith
- Choicetheorem's Overhauled Village
- Explorify
- The lost Castle
- Repurposed Structures
- Structuory 
- Structuory: Towers
- Nullscape
- Amplified Nether
- When Dungeons Arise
- Almost-Unified
- Dungeons and Taverns
- Gigeresque
- Towns and Towers
- Missing Wilds
- Applied Energistics 2
- Immersive Weathering

**Comment by Builderb0y** [comment]
The documentation got moved from the "scripting docs" folder to the [docs](https://github.com/Builderb0y/BigGlobe/tree/scriptable-generators/docs) folder. 3.x and MC 1.19.x are no longer maintained or supported, as things have changed a *lot* since then. You will need to update to a more modern MC version to make use of the documentation I've written, or reverse-engineer the old schemas from the built-in data files.

**Comment by DurekuRyuu** [comment]
Thanks, I look forward to trying it when Hexcasting gets updated past 1.19.2, in the meantime I will make do.
